YarnSpinner/Value.cs

```diff
@@ -0,0 +1,334 @@
+
+using System;
+
+namespace Yarn
+{
+	// A value from inside Yarn.
+	public class Value : IComparable, IComparable<Value> {
+        public static readonly Value DEFAULT = new Value();
+        
+		public enum Type {
+            Undefined = 0, // we don't know yet or don't have a value
+			Number,  // a constant number
+			String,  // a string
+			Bool,    // a boolean value
+			Variable, // the name of a variable; will be expanded at runtime
+			Null,    // the null value
+		}
+
+		public Value.Type type { get; internal set; }
+        
+        // this is primarily a debugging property, allowing
+        // for testing between actual and perceived type
+        internal object rawBacking { get; set; }
```

Comment:
I'll come clean here: I added that exclusively as a debug variable initially (to ensure the input to the constructor resulted in good state thruout the rest of the class), then made the mistake of leveraging against it in `Equals` and `GetHashCode`. I'll redo the latter and pull the `rawBacking` prop.